### DISCUSSION TITLE
drawing outside of canvas with raster brushes

### DISCUSSION THREAD

**Discussion by goldstargloww** [keiwando/toonsquid#339] [Ideas]
so i draw with pixel brushes normally, and i like being able to draw outside the canvas - its useful for a lot of things, like moving the camera or even just knowing what it looks like outside of frame to make sense of things, but this isn't possible with raster brushes, only with vector brushes, and this is _reallyyy_ annoying lol

i would use vector brushes, and have been to be honest, but there's a lot of things that're annoying about them and i'd really rather work with pixel brushes or otherwise raster brushes most of the time

i know i could make the canvas bigger, but that doesn't really solve the problem, just pushes them back a step - the camera might need to move beyond those boundaries or i might otherwise still need more space. this isnt a problem with vector brushes which only makes it all the more frustrating when it comes to raster brushes :/

**Comment by invadergrim** [comment]
THIS FR

**Comment by cvanalphen** [comment]
That would be really great to have. Procreate Dreams has this functionality too…..  I often use a clipping mask as rasterized  layer and add motion to it using keyframes but if I go off canvas, it gets automatically cropped 😢

**Reply by keiwando** [reply]
In theory this is undoubtedly a fantastic feature to have from a user's perspective, but the practical consequences of it are that it brings an enormous amount of complexity to the implementation of almost the entire app, including both existing and potential future features.
Procreate Dreams only "has" this feature with a lot of caveats, one of the biggest ones being that the brush strokes drastically degrade in quality if you draw them while zoomed out. The difference is already very noticeable between brush strokes drawn with the timeline expanded vs. with the timeline collapsed. I'd be curious to know what you think of this. I've heard from a lot of people that they find it unacceptable.

**Comment by cvanalphen** [comment]
Thank you for your reply Keiwando! 

We already have the possibility in Toonsquid to zoom in a rasterized layer... The degradation of the quality is there anyway and this issue applies to all rasterized drawing/photo software... Some apps can handle this problem better than others like Photoshop which has an option to increase the quality... For now, as a workaround, I create an animation clip so that I can manipulate it outside the canvas so that's already a good option :-)
